### PR TITLE
Adding in CSS and HTML linting. Fixes 72

### PR DESCRIPTION
**Pull request by jonathanKingston**

### CODE REVIEW

File: webextension/popup.html [hunk 1]
```diff
@@ -106,25 +106,25 @@ <h3 class="panel-header-text">Edit Containers</h3>
       <div class="small-10 columns">
         <form>
           <p>
-            <label>Name</label>
+            <label for="edit-container-panel-name-input">Name</label>
             <input type="text" name="container-name" id="edit-container-panel-name-input"/>
           </p>
           <p>
-            <label>Choose a color</label>
+            <label for="edit-container-panel-choose-color">Choose a color</label><!-- TODO this needs moving to use radio --!>
```
Comment (groovecoder) [inline webextension/popup.html]:
```
> testpilot-containers@0.0.1 lint:html /Users/lcrouch/code/testpilot-containers
> htmllint webextension/*.html

webextension/popup.html: line 113, col 13, an element with the id "edit-container-panel-choose-color" does not exist (should match `for` attribute)

[htmllint] found 1 errors out of 1 files
```

Comment (groovecoder) [inline webextension/popup.html]: `<div>` is not a [labelable Form element](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Content_categories#Form_labelable).

Comment (groovecoder) [inline webextension/popup.html]: And `--!>` needs to be `-->` or the following HTML is all commented out.

File: .htmllintrc [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "plugins": [],
+
+  "id-class-style": "dash",
+  "indent-style": "spaces",
+  "indent-width": 2,
+  "attr-name-ignore-regex": "http-equiv"
```
Comment (groovecoder) [inline .htmllintrc]: (After https://github.com/mozilla/testpilot-containers/pull/110/files#r98092302 is addressed), this needs to be `"attr-name-ignore-regex": "http-equiv|data.*"` or we get:
```
webextension/popup.html: line 116, col 63, attribute names must match the format: lowercase
webextension/popup.html: line 116, col 91, attribute names must match the format: lowercase
webextension/popup.html: line 117, col 63, attribute names must match the format: lowercase
webextension/popup.html: line 117, col 91, attribute names must match the format: lowercase
webextension/popup.html: line 118, col 63, attribute names must match the format: lowercase
webextension/popup.html: line 118, col 91, attribute names must match the format: lowercase
webextension/popup.html: line 119, col 63, attribute names must match the format: lowercase
webextension/popup.html: line 119, col 91, attribute names must match the format: lowercase
webextension/popup.html: line 120, col 63, attribute names must match the format: lowercase
webextension/popup.html: line 120, col 91, attribute names must match the format: lowercase
webextension/popup.html: line 121, col 63, attribute names must match the format: lowercase
webextension/popup.html: line 121, col 91, attribute names must match the format: lowercase
webextension/popup.html: line 122, col 63, attribute names must match the format: lowercase
webextension/popup.html: line 122, col 91, attribute names must match the format: lowercase
webextension/popup.html: line 123, col 63, attribute names must match the format: lowercase
webextension/popup.html: line 123, col 91, attribute names must match the format: lowercase
```

**Comment by pdehaan** [comment]
> Out of curiosity - are any other Mozilla/Firefox code-bases using these particular linting tools or styles?

I believe we used htmllint for some part of FxA l10n linting. I haven't used stylelint before, we usually use sass-lint, but that wasn't working for me due to the use of CSS variables.

File: webextension/js/popup.js [hunk 3]
```diff
@@ -432,27 +440,46 @@ Logic.registerPanel(P_CONTAINER_EDIT, {
     });
   },
 
+  initializeRadioButtons() {
+    const colorRadioTemplate = (containerColor) => {
+      return `<input type="radio" value="${containerColor}" name="container-color" id="edit-container-panel-choose-color-${containerColor}" />
+     <label for="edit-container-panel-choose-color-${containerColor}" class="usercontext-icon choose-color-icon" data-identity-icon="circle" data-identity-color="${containerColor}">`;
+    };
+    const colors = ["blue", "turquoise", "green", "yellow", "orange", "red", "pink", "purple" ];
```
Comment (groovecoder) [inline webextension/js/popup.js]: Can we pull this definition to the top of `popup.js` like we did with `IDENTITY_COLORS` in `index.js` ?

File: webextension/js/popup.js [hunk 4]
```diff
@@ -432,27 +440,46 @@ Logic.registerPanel(P_CONTAINER_EDIT, {
...
+    const colorRadioFieldset = document.getElementById("edit-container-panel-choose-color");
```
Comment (groovecoder) [inline webextension/js/popup.js]: Did we decide we like to use `let` when we're going to call methods on the object. (E.g., `appendChild` below)

File: webextension/js/popup.js [hunk 5]
```diff
@@ -432,27 +440,46 @@ Logic.registerPanel(P_CONTAINER_EDIT, {
...
+    colors.forEach((containerColor) => {
+      const templateInstance = document.createElement("span");
```
Comment (groovecoder) [inline webextension/js/popup.js]: Ditto `let` again?

File: webextension/js/popup.js [hunk 6]
```diff
@@ -432,27 +440,46 @@ Logic.registerPanel(P_CONTAINER_EDIT, {
...
+      templateInstance.innerHTML = colorRadioTemplate(containerColor);
+      colorRadioFieldset.appendChild(templateInstance);
+    });
+
+    const iconRadioTemplate = (containerIcon) => {
+      return `<input type="radio" value="${containerIcon}" name="container-icon" id="edit-container-panel-choose-icon-${containerIcon}" />
+     <label for="edit-container-panel-choose-icon-${containerIcon}" class="usercontext-icon choose-color-icon" data-identity-color="grey" data-identity-icon="${containerIcon}">`;
+    };
+    const icons = ["fingerprint", "briefcase", "dollar", "cart", "circle"];
```
Comment (groovecoder) [inline webextension/js/popup.js]: Ditto moving this to a top-line constant like `IDENTITY_ICONS` in `index.js`?

File: webextension/js/popup.js [hunk 7]
```diff
@@ -432,27 +440,46 @@ Logic.registerPanel(P_CONTAINER_EDIT, {
...
+    });
+  },
+
   // This method is called when the panel is shown.
   prepare() {
     const identity = Logic.currentIdentity();
     document.querySelector("#edit-container-panel-name-input").value = identity.name || "";
-
-    // FIXME: color and icon must be set. But we need the UI first.
+    [...document.querySelectorAll("[name='container-color']")].forEach(colorInput => {
```
Comment (groovecoder) [inline webextension/js/popup.js]: Neat use of spread.

**Comment by groovecoder** [comment]
Some comments, but none are blocking. Nice changes to a radio input fieldset, and it works well.

**Pull request merged**